### PR TITLE
OCPQE-13470: Build quay test image to run golang code

### PR DESCRIPTION
**Pull request by xiaozhisean**

### CODE REVIEW

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse pull-ci-quay-quay-tests-master-quay-upgrade-images

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse pull-ci-quay-quay-tests-master-quay-upgrade-images

File: ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    namespace: ocp
+    tag: rhel-8-golang-1.20-openshift-4.16
+images:
+- dockerfile_path: images/Dockerfile.upgrade
+  from: tools
+  inputs:
+    ocp_builder_rhel-8-golang-1.20-openshift-4.16:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.16
+  to: quay-test-upgrade
```
Comment (LiZhang19817) [inline ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-upgrade.yaml:26]: As this image will be used for Quay Operator testing with golang code, not only for upgrade, suggest use name like "quay-test-operator"

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse pull-ci-quay-quay-tests-master-quay-upgrade-images

File: ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-upgrade.yaml (hunk 1 above)
Comment (SeanZhao-redhat) [reply ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-upgrade.yaml]: updated

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse pull-ci-quay-quay-tests-master-quay-upgrade-images

**Pull request merged**